### DISCUSSION TITLE
weird visual mode behavior when I map i to k

### DISCUSSION THREAD

**Discussion by Jiyangson** [LazyVim/LazyVim#1098] [Q&A]
I want to remap navigation keys as follows
```
local keymap = vim.keymap

keymap.set({ "n", "v" }, "h", "i", { noremap = true })
keymap.set({ "n", "v" }, "j", "h", { noremap = true })
keymap.set({ "n", "v" }, "k", "j", { noremap = true })
keymap.set({ "n", "v" }, "i", "k", { noremap = true })
```
It works in normal mode, but in visual mode, sometimes which-key windows pops up when I press i and in visual line mode, it lags when I press i.

and whichkey do have a warning about this
![image](https://github.com/LazyVim/LazyVim/assets/31755447/2b427420-873f-4ba3-808d-82c0533f5eb7)

How can I fix this? Thanks a lot

**Comment by dpetka2001** [comment]
That's because `i` in `x` (Visual mode) is mapped as `inner textobject` from `mini.ai`, so the lag you're experiencing is because it waits for a textobject to follow. Check `verbose vmap i` to see for yourself. Also check [mini.ai docs](https://github.com/echasnovski/mini.ai) the default config. Maybe you would like to change the internal mapping of `mini.ai` to something else (in `mappings`)? Haven't tried it, so don't know if it works. But I think you would also have to change the config [Lazyvim](https://www.lazyvim.org/plugins/coding#miniai) sets for `mini.ai` as well because it also sets `which-key` keymaps for `mini.ai`.
